fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -7335,23 +7339,55 @@ public void onEraseOlapTable(OlapTable olapTable, boolean isReplay) {
                     for (Tablet tablet : materializedIndex.getTablets()) {
                         long tabletId = tablet.getId();
                         if (partition.isUseStarOS()) {
-                            DropReplicaTask dropTask = new DropReplicaTask(
-                                    ((StarOSTablet) tablet).getPrimaryBackendId(), tabletId, schemaHash, true);
+                            long backendId = ((StarOSTablet) tablet).getPrimaryBackendId();
+                            DropReplicaTask dropTask = new DropReplicaTask(backendId, tabletId, schemaHash, true);
+                            AgentBatchTask batchTask = batchTaskMap.get(backendId);
+                            if (batchTask == null) {
+                                batchTask = new AgentBatchTask();
+                                batchTaskMap.put(backendId, batchTask);
+                            }
                             batchTask.addTask(dropTask);
                         } else {
                             List<Replica> replicas = ((LocalTablet) tablet).getReplicas();
                             for (Replica replica : replicas) {
                                 long backendId = replica.getBackendId();
                                 DropReplicaTask dropTask = new DropReplicaTask(backendId, tabletId, schemaHash, true);
+                                AgentBatchTask batchTask = batchTaskMap.get(backendId);
+                                if (batchTask == null) {
+                                    batchTask = new AgentBatchTask();
+                                    batchTaskMap.put(backendId, batchTask);
+                                }
                                 batchTask.addTask(dropTask);
                             } // end for replicas
                         }
                     } // end for tablets
                 } // end for indices
             } // end for partitions
-            AgentTaskExecutor.submit(batchTask);
-        }
 
+            int numDropTaskPerBe = Config.max_agent_tasks_send_per_be;
+            for (Map.Entry<Long, AgentBatchTask> entry : batchTaskMap.entrySet()) {
```

Comment:
Drop replica tasks can be executed parallel between backends and this can be a little faster.